internal/provider/helpers_test.go

```diff
@@ -0,0 +1,9 @@
+package provider
+
+import (
+	"strings"
+)
+
+func composeTestConfigFunc(configs ...string) string {
```

Comment:
Necessary? Having `strings.Join(, "\n")` below would still look pretty clean!

Reply:
The alternative is this which I guess is OK. I like the convenience of a helper function here

```
Config: strings.Join([]string{
  config1, config2, ...,
}, "\n"),
```